src/guide/customization.rs

```diff
@@ -0,0 +1,398 @@
+//! A tutorial on how to customize eyre by overriding the Context parameter
+//!
+//! # Introduction
+//!
+//! In this tutorial we will go through the process of implementing a custom context step by step:
```

Comment:
I think it'd be really valuable to have a section before you begin implementing a custom context explaining what a custom context is, and why you might want one. It doesn't necessarily have to be here, but a sentence mentioning where in the docs you can learn the motivation would be nice.